source/common/network/cidr_range.cc

```diff
@@ -0,0 +1,177 @@
+#include "cidr_range.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/common/assert.h"
+#include "common/common/utility.h"
+#include "common/network/address_impl.h"
+#include "common/network/utility.h"
+
+namespace Network {
+namespace Address {
+
+CidrRange::CidrRange() : length_(-1) {}
+
+CidrRange::CidrRange(const InstancePtr& address, int length) : address_(address), length_(length) {
```

Comment:
For shared_ptr just pass InstancePtr (no const&). This can lead to subtle bugs and the compiler will usually optimize.